NPE during spark submission

On the result of merging PR53 into our main branch:

```
Exception in thread "main" java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at org.spark_project.guava.util.concurrent.AbstractFuture$Sync.getValue(AbstractFuture.java:306)
	at org.spark_project.guava.util.concurrent.AbstractFuture$Sync.get(AbstractFuture.java:279)
	at org.spark_project.guava.util.concurrent.AbstractFuture.get(AbstractFuture.java:96)
	at org.apache.spark.deploy.kubernetes.Client$$anonfun$run$6$$anonfun$apply$5$$anonfun$apply$7.apply(Client.scala:189)
	at org.apache.spark.deploy.kubernetes.Client$$anonfun$run$6$$anonfun$apply$5$$anonfun$apply$7.apply(Client.scala:148)
	at org.apache.spark.util.Utils$.tryWithResource(Utils.scala:2530)
	at org.apache.spark.deploy.kubernetes.Client$$anonfun$run$6$$anonfun$apply$5.apply(Client.scala:148)
	at org.apache.spark.deploy.kubernetes.Client$$anonfun$run$6$$anonfun$apply$5.apply(Client.scala:133)
	at org.apache.spark.util.Utils$.tryWithResource(Utils.scala:2530)
	at org.apache.spark.deploy.kubernetes.Client$$anonfun$run$6.apply(Client.scala:133)
	at org.apache.spark.deploy.kubernetes.Client$$anonfun$run$6.apply(Client.scala:105)
	at org.apache.spark.util.Utils$.tryWithResource(Utils.scala:2530)
	at org.apache.spark.deploy.kubernetes.Client.run(Client.scala:105)
	at org.apache.spark.deploy.kubernetes.Client$.main(Client.scala:675)
	at org.apache.spark.deploy.kubernetes.Client.main(Client.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:750)
	at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:178)
	at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:203)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:117)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
Caused by: java.lang.NullPointerException
	at scala.Predef$.Boolean2boolean(Predef.scala:366)
	at org.apache.spark.deploy.kubernetes.Client$$anonfun$21.apply(Client.scala:598)
	at org.apache.spark.deploy.kubernetes.Client$$anonfun$21.apply(Client.scala:598)
	at scala.collection.TraversableLike$$anonfun$filterImpl$1.apply(TraversableLike.scala:248)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$class.filterImpl(TraversableLike.scala:247)
	at scala.collection.TraversableLike$class.filterNot(TraversableLike.scala:267)
	at scala.collection.AbstractTraversable.filterNot(Traversable.scala:104)
	at org.apache.spark.deploy.kubernetes.Client.org$apache$spark$deploy$kubernetes$Client$$buildDriverSubmissionClient(Client.scala:598)
	at org.apache.spark.deploy.kubernetes.Client$DriverPodWatcher.eventReceived(Client.scala:389)
	at org.apache.spark.deploy.kubernetes.Client$DriverPodWatcher.eventReceived(Client.scala:329)
	at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager$1.onMessage(WatchConnectionManager.java:216)
	at okhttp3.internal.ws.RealWebSocket$1.onMessage(RealWebSocket.java:62)
	at okhttp3.internal.ws.WebSocketReader.readMessageFrame(WebSocketReader.java:242)
	at okhttp3.internal.ws.WebSocketReader.processNextFrame(WebSocketReader.java:108)
	at okhttp3.internal.ws.RealWebSocket.readMessage(RealWebSocket.java:97)
	at okhttp3.ws.WebSocketCall.createWebSocket(WebSocketCall.java:152)
	at okhttp3.ws.WebSocketCall.access$000(WebSocketCall.java:41)
	at okhttp3.ws.WebSocketCall$1.onResponse(WebSocketCall.java:97)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:126)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
```

I think this diff will fix it:

```
diff --git a/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala b/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala
index 5c86512..142e1b9 100644
--- a/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala
+++ b/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala
@@ -595,7 +595,8 @@ private[spark] class Client(
       .filter(_.getName == SUBMISSION_SERVER_PORT_NAME)
       .head.getNodePort
     val nodeUrls = kubernetesClient.nodes.list.getItems.asScala
-      .filterNot(_.getSpec.getUnschedulable)
+      .filterNot(node => node.getSpec.getUnschedulable != null &&
+        node.getSpec.getUnschedulable)
       .flatMap(_.getStatus.getAddresses.asScala.map(address => {
         s"$urlScheme://${address.getAddress}:$servicePort"
       })).toArray
```

Doing some testing now..  cc @mccheah

--- a/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala
+++ b/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala
@@ -570,7 +570,8 @@ private[spark] class Client(
       .filter(_.getName == SUBMISSION_SERVER_PORT_NAME)
       .head.getNodePort
     val nodeUrls = kubernetesClient.nodes.list.getItems.asScala
-      .filterNot(_.getSpec.getUnschedulable)
+      .filterNot(node => node.getSpec.getUnschedulable != null &&
+        node.getSpec.getUnschedulable)
       .flatMap(_.getStatus.getAddresses.asScala.map(address => {
         s"$urlScheme://${address.getAddress}:$servicePort"
       })).toArray

